core/src/test/scala/org/broadinstitute/dsde/rawls/spendreporting/SpendReportingServiceSpec.scala

```diff
@@ -60,7 +60,7 @@ class SpendReportingServiceSpec extends AnyFlatSpecLike with TestDriverComponent
       )
       val page: PageImpl[FieldValueList] = new PageImpl[FieldValueList](null, null, fieldValues.asJava)
       val schema: Schema = Schema.of(fields:_*)
-      val tableResult: TableResult = new TableResult(schema, 2, page)
+      val tableResult: TableResult = new TableResult(schema, fieldValues.length, page)
```

Comment:
I'm getting a bit concerned about how big this test class is getting. A lot of fake test data and different assertions are required for the more complicated aggregations. We could work some magic with inheritance to slim this down, but this PR is already a somewhat significant refactor so I didn't want to do anything too drastic here. Let me know if you disagree and I can see about cutting this class down in size!

Reply:
This is ok with me, test cases for complex reporting logic like this can be somewhat gnarly.

Reply:
Where we are asserting the same data in multiple test points (aggregation by `Compute`, aggregation by `Workspace~Compute`), I think it would really help to pull those blocks into helper methods that the individual test points can call. With well-named helper methods, the test points will become much easier to read, and it removes the code duplication.

Reply:
+1 on this.